### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by javiertejedorm** [llvm/llvm-project#134]
Hello,
I am calling to the function `__sanitizer_symbolize_pc` several times from my application and it is leaking memory.

I have executed also your tests inside an endless loop and they are also leaking memory:

https://github.com/llvm/llvm-project/blob/master/compiler-rt/test/sanitizer_common/TestCases/symbolize_pc.cpp

```
#include <sanitizer/common_interface_defs.h>

int main() {

    void *PC = __builtin_return_address(0);
    char PcDescr[1024];

    for(;;){

        __sanitizer_symbolize_pc(PC, "%p %F %L", PcDescr, sizeof(PcDescr));

    }

    return 0;
}
```

I also detected `__sanitizer_symbolize_global`. I haven't tested the rest of the API.

Is there any other way that I can use to avoid this problem? I need to extract the function and the source file from the PC.

Thanks.

**Comment by androm3da** [comment]
It's entirely possible that you have found a defect but how do you know there's a leak?

**Comment by javiertejedorm** [comment]
Thanks for the replay. I'm running it in Ubuntu. I am monitoring the process with System Monitor (or `top` command) and I see that the memory consumed by the process increases without stop.

**Comment by androm3da** [comment]
You are looking at the RSS and it is growing unbounded?  Even still this could be a confusing signal. As long as the memory is released before the program exits, it's not a leak. 

Generally I'd advise that you use LeakSanitizer to check but it's not clear to me whether you could safely do that with sanitizer code itself. 

Maybe it would be better to discuss what led you to start looking there?

**Comment by javiertejedorm** [comment]
Yes, I am looking at the Memory field of `top` and it grows unbounded (until the system starts consuming the swap memory and hangs)
I am using LeakSanitizer and it does not detect any leak.
The memory could be released when it exits but my program does not exit, the user decides it.
I think, after returning from `__sanitizer_symbolize_pc` the memory should be already released.
